Frameworks/Foundation/NSScanner.mm

```diff
@@ -268,62 +234,93 @@ - (BOOL)scanFloat:(float*)valuep {
  @Status Interoperable
 */
 - (BOOL)scanDouble:(double*)valuep {
-    char* pScanStart = (char*)[_string UTF8String];
-    char* pScanEnd = NULL;
+    NSUInteger startLocation = [self _indexOfNextUnskippedCharacter];
+    const NSUInteger length = _string.length;
+    if (startLocation >= length) {
+        return NO;
+    }
 
-    pScanStart += _location;
+    char* pScan = (char*)[_string UTF8String] + startLocation;
 
-    double val = strtod(pScanStart, &pScanEnd);
-    assert(pScanEnd != NULL);
+    // strtod automatically takes care of spaces (needs to be explicitly specified in charactersToBeSkipped),
+    // and only expects '.' as a decimal separator
+    // copy a scannable range of the string to a new buffer, adjusting for these constraints
+    char decimalSeparator;
+    if ((!_locale) ||
+        (![[_locale objectForKey:NSLocaleDecimalSeparator] getCString:&decimalSeparator maxLength:1 encoding:NSASCIIStringEncoding])) {
```

Comment:
>NSASCIIStringEncoding [](start = 109, length = 21)

In some locales the decimal separator won't be representable as an ASCII character.